bayesianapproximator.py

```diff
@@ -7,47 +7,39 @@ def __init__(self, state_dimensions, num_acts):
 
         self.dimensions = np.array(state_dimensions)
         self.num_actions = num_acts
-    #     self.B = np.zeros(np.prod(state_dimensions) + [num_acts])
 
     def update_stats(self, s, a, val=0.0):
         raise NotImplementedError
 
     def sample(self, s, a, n):
         raise NotImplementedError
 
-
-
 class TabularBayesianApproximation(BayesianApproximator):
   def __init__(self, state_dimensions, num_acts):
-    super().__init__(state_dimensions, num_acts)
-    self.B = np.zeros([np.prod(self.dimensions)] + [self.num_actions, 4])
-    print("Initiating tabular bayes with %d states and %d actions"%(np.prod(state_dimensions), num_acts))
-
+    num_states = np.prod(state_dimensions)
+    self.state_shape = state_dimensions
+    self.B = np.zeros((num_states, num_acts, 4))
     # prior sample mean
     # prior "observations to make that mean"
     # prior "observations to make our variance" # try to make it hard to reduce this
     # prior sum of square errors (proportional to initial sample variance)
-    self.B[:, :] = [1, 1, 1, 4]
+    self.B[:, :] = [1, 1, 2, 1]
```

Comment:
Fatima and I found these meta-parameters seem to be reasonable defaults (though we haven't thoroughly experimented).